### PR TITLE
Feature/migrate components directory to composition api

### PR DESCRIPTION
**Pull request by PeterThrels**

### CODE REVIEW

**Review by oleggrech-t — APPROVED** [review]

File: components/molecules/Header.vue [hunk 1]
```diff
@@ -21,24 +21,31 @@
...
+            await corporates?.get()
         }
     }
+})
+</script>
+
+<script lang="ts">
+export default {
+    name: 'HeaderComponent',
 }
```
Comment (gaucj-t) [inline components/molecules/Header.vue:49]: Do we need this ? we can simply use the component file name. In any case if we need to change the component name we should change the file name.

File: components/molecules/MobileComponent.vue [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+import ErrorAlert from '~/components/molecules/ErrorAlert.vue'
+import LoaderButton from '~/components/atoms/LoaderButton.vue'
+import useZodValidation from '~/composables/useZodValidation'
+
+export default defineComponent({
+    components: {
+        ErrorAlert,
+        LoaderButton,
+    },
+    layout: 'auth',
```
Comment (gaucj-t) [inline components/molecules/MobileComponent.vue:105]: not sure why we need this but its ok.

File: components/organisms/accounts/statement/statement.vue [hunk 3]
```diff
@@ -63,102 +63,121 @@
...
+import { useFilters } from '~/composables/useFilters'
+import { useRouterFilter } from '~/composables/useRouterFilter'
 import { GetManagedAccountStatementRequest } from '~/plugins/weavr-multi/api/models/managed-instruments/managed-account/requests/GetManagedAccountStatementRequest'
-import AccountsMixin from '~/mixins/AccountsMixin'
 import { OrderEnum } from '~/plugins/weavr-multi/api/models/common/enums/OrderEnum'
 
-@Component({
+export default defineComponent({
+    name: 'AccountStatementComponent',
+    // TODO: Update this after nuxt bridge
```
Comment (gaucj-t) [inline components/organisms/accounts/statement/statement.vue:78]: we can remove this an use setup tag i think. We can then just rename the file name to account statements and refactor imports where necessary. then we can use setup tag.

File: components/organisms/accounts/statement/statement.vue [hunk 4]
```diff
@@ -63,102 +63,121 @@
...
-    downloadStatement() {
-        const _routeQueries = dot.object(this.$route.query)
-        const _filters = _routeQueries.filters ? _routeQueries.filters : {}
+        // TODO: Resolve to useBase Composable
+        // Move all luxon function calls to utils for reusability
+        const formatDate = (val) => {
+            const dateTime = DateTime.fromJSDate(val)
+            if (dateTime.hasSame(DateTime.now(), 'year')) {
+                return dateTime.toFormat('d MMMM')
+            }
```
Comment (gaucj-t) [inline components/organisms/accounts/statement/statement.vue:132]: i think we already did a ticket for this correct ?

File: components/organisms/cards/statement/statement.vue [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        DownloadIcon: () => import('~/assets/svg/download.svg?inline'),
+        DeleteIcon: () => import('~/assets/svg/delete.svg?inline'),
+        StatementItem: () => import('~/components/organisms/StatementItem.vue'),
+    },
+    props: {
+        filters: {
+            type: Object as PropType<GetManagedCardStatementRequest>,
+            default: null,
+        },
+    },
```
Comment (gaucj-t) [inline components/organisms/cards/statement/statement.vue:97]: same here use setup tag if possible.

**Review by gaucj-t — CHANGES_REQUESTED** [review]

File: components/molecules/Header.vue (hunk 1 above)
Comment (PeterThrels) [reply components/molecules/Header.vue:49]: Yeah, there was a lint error, we can change the component name.

File: components/organisms/accounts/statement/statement.vue (hunk 3 above)
Comment (PeterThrels) [reply components/organisms/accounts/statement/statement.vue:78]: The issue here is the way assets are imported, I think we're blocked by [this](https://weavr-payments.atlassian.net/browse/WEB-176) if not we have to use defineComponent()

File: components/organisms/accounts/statement/statement.vue (hunk 4 above)
Comment (PeterThrels) [reply components/organisms/accounts/statement/statement.vue:132]: Yes, done [here](https://weavr-payments.atlassian.net/browse/WEB-205)

File: components/organisms/cards/statement/statement.vue (hunk 5 above)
Comment (PeterThrels) [reply components/organisms/cards/statement/statement.vue:97]: Same as [here](https://github.com/weavr-io/onvirtual.cards/pull/189#discussion_r1607860177)

Comment (gaucj-t) [reply components/organisms/cards/statement/statement.vue:97]: ok

File: components/organisms/accounts/statement/statement.vue (hunk 4 above)
Comment (gaucj-t) [reply components/organisms/accounts/statement/statement.vue:132]: cool

File: components/molecules/Header.vue (hunk 1 above)
Comment (PeterThrels) [reply components/molecules/Header.vue]: done

**Review by jasnvella2 — APPROVED** [review]

**Review by gaucj-t — APPROVED** [review]

**Pull request merged**